music/src/core/visualizer.ts

```diff
@@ -450,7 +464,11 @@ export class PianoRollSVGVisualizer extends BaseVisualizer {
     rect.setAttribute('y', `${Math.round(y)}`);
     rect.setAttribute('width', `${Math.round(w)}`);
     rect.setAttribute('height', `${Math.round(h)}`);
-    rect.setAttribute('data-index', `${index}`);
+    dataAttributes.forEach(([key, value]: DataAttribute) => {
+      if (value !== undefined) {
+        rect.dataset[key] = `${value}`;
```

Comment:
this works even though `value` is a number because you are explicitely converting it to a string here. This is why I don't think you really need that `any` ...

Reply:
Maybe I'm missing something, but I still don't know what you want me to do instead. If I just do `type DataAttribute = [string, string]`, then TypeScript will make me convert to string in `draw()` rather than here. This also means I need to check for `undefined` in `draw` *as well as here*, i.e.:
```typescript
      const dataAttributes : DataAttribute[] = [
        ['index', i],
        ['instrument', note.instrument === undefined ? undefined : `${note.instrument}`],
        ...
      ];
```

In that case it would be easier to just `return rect` from here and then in `draw()`, do
```typescript
if (note.instrument !== undefined) rect.dataset.instrument = `${note.instrument}`;
...
```

Or, if you're OK with setting `data-instrument="undefined"`, then I don't need to check for it at all.

Reply:
Let's just leave it how it is, I think it's fine enough.